Add notes on testing, remove --runInBand

dcloud opened the pull request:
**Description of change**


**How to test**


**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/263

**Checklist**
<!-- Add details to each completed item -->
- [ ] Meets issue criteria
- [ ] Code tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

dcloud commented:
This is a light PR with some notes and minor changes, since the cause of backend tests failures was already addressed.

I also have a branch with some [complex](https://github.com/adhocteam/Head-Start-TTADP/blob/dcloud/split-docker-local-test/bin/docker-run) [changes](https://github.com/adhocteam/Head-Start-TTADP/blob/dcloud/split-docker-local-test/docker-compose.test.yml) that I would like to run by the team.

jasalisbury commented on docs/testing.md:
````diff
@@ -0,0 +1,31 @@
+Here's how that might look:
+
+```
+let a = await someAsyncFun();
````
In this example wouldn't `await someAsyncFun();` wait for the resulting `someAsyncFunc` promise to resolve? This would mean `a` is not a promise but the resolved value of the promise? Should this be `let a = someAsyncFunc()`?

dcloud replied:
Oh yeah, yup. I was going to write a before/after example, and then did neither. Good catch

jasalisbury reviewed (commented):
:tada: Looks good! Just one question about the example `Promise.all` usage!

jasalisbury reviewed (approved):
:+1:

kryswisnaskas reviewed (approved):
👍

dcloud commented:
After looking into some issues with @dcmcand, we added some changes that mitigate errors arising from tests deleting more database records than they create. I also added a section on that into the `docs/testing.md`.

cc @kryswisnaskas
